java/code/src/com/redhat/rhn/manager/system/entitling/SystemUnentitler.java

```diff
@@ -85,6 +95,10 @@ public void removeServerEntitlement(Server server, Entitlement ent) {
                 }
             });
         }
+
```

Comment:
while you are at it, maybe you can fix this part https://github.com/uyuni-project/uyuni/pull/2325/files?file-filters%5B%5D=.java#diff-8cc725b56315ba8fd87da99a53fb4fcdL76-R88 

The conditions here should be inverted, as in https://github.com/uyuni-project/uyuni/blob/master/java/code/src/com/redhat/rhn/manager/system/entitling/SystemEntitler.java#L128-L134

Reply:
not sure I really understood where you want that change, but I'll try to find it

Reply:
https://github.com/uyuni-project/uyuni/blob/master/java/code/src/com/redhat/rhn/manager/system/entitling/SystemUnentitler.java#L76-L78

This is wrong, and was introduced in https://github.com/uyuni-project/uyuni/commit/0d8fe90492c3bfd8620191a1d9c61af1b160f0f4

The right way is to do it as in https://github.com/uyuni-project/uyuni/blob/master/java/code/src/com/redhat/rhn/manager/system/entitling/SystemEntitler.java#L128-L134 , meaning check if the server is a minion, ,then update the groups pillar, and then the specific features like mointoring or virtualization

Reply:
Thanks for pointing it out, and sorry as it was my mistake!

Reply:
haha no worries!